target/linux/ramips/dts/mt7628an_joowin_jw-wr758ac.dts

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "joowin,jw-wr758ac", "mediatek,mt7628an-soc";
+
+	keys {
+		compatible = "gpio-keys";
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 38 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wifi-high {
+			label = "blue:wifi-high";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi-med {
+			label = "blue:wifi-med";
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi-low {
+			label = "blue:wifi-low";
+			gpios = <&gpio 37 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&state_default {
+	gpio {
+		groups = "i2c", "wled_an", "gpio", "refclk", "wdt", "p0led_an";
+		function = "gpio";
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	mt76@0,0 {
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+		ieee80211-freq-limit = <5000000 6000000>;
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x50000 0x760000>;
```

Comment:
What is after 0x7b0000?

Reply:
This is as it is on the vendor firmware (see https://forum.openwrt.org/t/adding-support-for-comfast-cf-wr758ac-aka-joowin-jw-wr758ac/87355/21 for full dmesg output, as it's also possible to modify the vendor firmware and re-flash it in order to obtain root ssh access).

From vendor dmesg, the part that interests is:
```
[    0.690000] Creating 5 MTD partitions on "spi32766.0":
[    0.700000] 0x000000000000-0x000000030000 : "u-boot"
[    0.710000] 0x000000030000-0x000000040000 : "u-boot-env"
[    0.710000] 0x000000040000-0x000000050000 : "factory"
[    0.720000] 0x000000050000-0x0000007b0000 : "firmware"
```

And the /proc/mtd in vendor firmware after first boot:
```
root@JOOWIN:~# cat /proc/mtd 
dev:    size   erasesize  name
mtd0: 00030000 00010000 "u-boot"
mtd1: 00010000 00010000 "u-boot-env"
mtd2: 00010000 00010000 "factory"
mtd3: 00760000 00010000 "firmware"
mtd4: 0012386d 00010000 "kernel"
mtd5: 0063c793 00010000 "rootfs"
mtd6: 00110000 00010000 "rootfs_data"
mtd7: 00040000 00010000 "configs"
```

What are your recommendations?

Reply:
If the flash memory area after 0x7b0000 is indeed empty, then you might be able to extend the firmware partition.
If not (or to be safe), keep it as is. In that case you can add a note that 0x7b0000~0x800000 is intentionally left unused.

Reply:
Fixed in latest push, firmware partition now goes up to 0x800000. Tested with success.